adr/adr.001.folder structure/adr.001.folder structure.md

```diff
@@ -0,0 +1,260 @@
+  b. If so, move it to the appropriate feature directory.
+  c. If not, consider moving it to the `shared/` directory.
+- **Step 5:** Update import paths for moved files and their
+  dependencies.
+- **Step 6:** Gradually refactor existing features during
+  planned refactoring sprints or as part of regular
+  maintenance.
+- **Step 7:** Regularly review and update documentation to
+  reflect the evolving folder structure.
+- **Step 8:** Conduct team training sessions to ensure all
+  developers understand and follow the new structure.
+
+## File Naming Conventions
+
+We will adopt a consistent naming convention for key files
+within each feature directory. This approach makes it easy
+to search for and identify specific file types, and clearly
+indicates the purpose of each file.
+
+- **Key File Types:**
+  - `[featureName].screen.tsx`: Feature screens
+  - `[featureName].nav.tsx`: Navigation components
+  - `[featureName].store.ts`: Data stores
+  - `[featureName].queries.ts`: API queries - using React Query
+  - `[featureName].utils.ts`: Utility functions
+  - `[featureName].data.ts`: Data models or types
+  - `[featureName].actor.ts`: State machines or actors
+  - `[featureName].client.ts`: Clients that control dependencies
+  - `[featureName].types.ts`: Types for this particular feature
+
+This naming convention offers several benefits:
+1. Easy to search: Use command+P to quickly find all 
+   relevant files for a particular domain.
+2. Clear purpose: File names immediately indicate their 
+   function within the feature.
+3. Consistent organization: Standardizes file naming 
+   across the project.
+
+*Example of the new structure for the Accounts feature:*
+
+```
+features/
+└── accounts/
+    ├── components/
+    │   ├── AccountSettingsButton.tsx
+    │   ├── ExternalWalletPicker.tsx
+    │   ├── EphemeralAccountBanner.tsx
+    │   ├── ProfileSettingsButton.tsx
+    │   └── Logout/
+    │       ├── privy.tsx
+    │       ├── privy.web.tsx
+    │       ├── wallet.tsx
+    │       └── wallet.web.tsx
+    ├── screens/
+    │   ├── Accounts.screen.tsx
+    │   ├── AccountsAndroid.screen.tsx
+    │   └── Onboarding/
+    │       ├── Onboarding.screen.tsx
+    │       ├── Onboarding.web.screen.tsx
+    │       └── components/
+    ├── navigation/
+    │   ├── Accounts.nav.tsx
+    │   ├── AccountsDrawer.nav.tsx
+    │   └── OnboardingNav.nav.tsx
+    ├── hooks/
+    │   ├── useDisconnectActionSheet.ts
+    │   └── usePhotoSelect.ts
+    ├── utils/
+    │   ├── addressBook.ts
+    │   ├── keychain/
+    │   ├── wallet.ts
+    │   └── evm/
+    ├── data/
+    │   └── store/
+    │       └── accounts.store.ts 
+    ├── accounts.utils.ts
+    ├── accounts.client.ts
+    └── accounts.queries.ts
+
+
+*Note:* Care should be taken to ensure that any platform-specific files (e.g., `.ios.tsx`, `.android.tsx`, `.web.tsx`) are correctly placed and that conditional imports are handled properly.
+
+## Migration Plan
```

Comment:
_proposed_ migration plan

Reply:
@thierryskoda I'm going to copy/paste [your feedback from slack](https://xmtp-labs.slack.com/archives/C07NSHXK693/p1729109383162139?thread_ts=1729107402.644529&cid=C07NSHXK693)

> Watched the video and gave some comments. Super excited about this! I'm 100% aligned with the proposed structure.
> What will be the hardest is how to all sync ourselves while making folder/files changes.

Could you review the plan below and see if that tracks to you?

> I think we should do them isolated yes but QUICKLY so we don't accumulate conflicts...

Absolutely. Care needs to be taken to not both be doing something and move/change files someone else is working on. Moving the files in one PR and changing in another is my proposed solution. 

> So the PR should be ONLY changing structure and merged and then we can fix other stuff.

One PR to move files. Get that PR merged. THEN change files. 👍